### ISSUE TITLE
[Bug] Font issue on first letter of appended components when changing colors.

### ISSUE THREAD

**Issue by lanAnderson** [BloCamLimb/ModernUI#234]
### **Describe the bug**
When appending two component that have different fonts, changing the color of the second component causes its first letter to use the font from the first component. 

-----------------------------

### **To Reproduce/Expected behavior/Screenshots**
  Working as expected:

  ![2024-08-29_06 13 38](https://github.com/user-attachments/assets/5da254ef-8230-486e-b3a6-20dde95ee367)

```
      MutableComponent component1 = Component.literal("Test1 ")
          .withStyle(Style.EMPTY.withFont(Minecraft.ALT_FONT));

      MutableComponent component2 = Component.literal("Test2")
          .withStyle(Style.EMPTY.withFont(Minecraft.DEFAULT_FONT));
      
      MutableComponent finalComponent = component1.append(component2);

      mc.font.drawShadow(graphics, finalComponent, 10, 10, 0xffffff);
```

  Adding a color to second component causes its first letter to continue using the font of the first component:

  ![2024-08-29_06 13 58](https://github.com/user-attachments/assets/47771e35-7016-49fd-9835-fa76040566bd)

```
      MutableComponent component1 = Component.literal("Test1 ")
          .withStyle(Style.EMPTY.withFont(Minecraft.ALT_FONT));

      MutableComponent component2 = Component.literal("Test2")
          .withStyle(Style.EMPTY.withFont(Minecraft.DEFAULT_FONT).withColor(ChatFormatting.RED));

      MutableComponent finalComponent = component1.append(component2);

      mc.font.drawShadow(graphics, finalComponent, 10, 10, 0xffffff);
```
-----------------------------

### **Environment**
 - Mod Version: 3.10.1.2
 - Forge Version: 43.4.2 (Tested on 1.20.1 Forge as well)
 - Operating System: Windows 11
 - Graphics Card: RTX 4070 Ti
 - Other Mods: Tested with just Modern UI
-----------------------------
### **Other Info**
Originally reported here: https://github.com/IanMods/SeasonHUD/issues/109

I basically use a custom bitmap font for icons, and when the text color is changed it cause the first letter of the text appended to the icon to use the incorrect font.

**Comment by BloCamLimb** [comment]
Fixed in https://github.com/BloCamLimb/ModernUI-MC/commit/011cb5adea7f240046faf2beeb95d066d642d76f

**Comment by BloCamLimb** [comment]
Fixed in 3.11
